luigi/scheduler.py

```diff
@@ -822,7 +921,22 @@ def get_work(self, host=None, assistant=False, current_tasks=None, worker=None,
                  'task_id': None,
                  'n_unique_pending': n_unique_pending}
 
-        if best_task:
+        if len(batched_tasks) > 1:
+            batch_id = hashlib.md5('|'.join(task.id for task in batched_tasks)).hexdigest()
+            for task in batched_tasks:
+                self._state.set_batch_running(task, batch_id, worker_id)
+
+            combined_params = best_task.params.copy()
+            combined_params.update(batched_params)
```

Comment:
batched_params couldn't be None here because batched_tasks gets filled in only when batched_params is filled in. If we managed to get here with an empty batched_params, that's a huge logic failure in the scheduler too.